app.py

```diff
@@ -1,67 +1,129 @@
+    except:
+        context = {
+            'error': True,
+            'title': weather_data['title'],
+            'description': weather_data['detail']
+        }
+        return render_template('index.html', **context)
 
-    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state']
+    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state'] +['',' (Default)'][default_forecast]
 
     context = {
         'timestamp': timestamp,
         'periodic_data': periodic_data,
         'areaDescription': areaDescription
     }
 
-    return render_template('index.html', **context)
+    return context
+
+@app.route('/')
+@app.route('/index')
+@app.route('/<zip_code>')
+def index(zip_code=None):
+    
+    timestamp = datetime.datetime.now()
+
+    if zip_code != None and re.search(r'.*(\d{5}?)$', zip_code):
+        if cache.get(CACHE_1_ZIP) is not None and cache.get(CACHE_1_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_1_ZIP)['context'])
+        elif cache.get(CACHE_2_ZIP) is not None and cache.get(CACHE_2_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_2_ZIP)['context'])
+        elif cache.get(CACHE_3_ZIP)  is not None and cache.get(CACHE_3_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_3_ZIP)['context'])
+        else:
+            lat, lng, default_value = 0, 0, False
+            
+            # zipcode-lat-lng.csv
+            # https://gist.github.com/abatko/ee7b24db82a6f50cfce02afafa1dfd1e
+            with open('./assets/zipcode-lat-lng.csv', 'r') as f:
+                for line in f:
+                    data = line.split(',')
+                    if data[0] == zip_code:
+                        lat, lng = data[1], data[2].rstrip('\n')
+                        break
+            
+            if lat == 0 and lng == 0:
+                context = {
+                    'error': True,
+                    'title': 'Something went wrong',
+                    'description': 'We couldn\'t find information for the zipcode %s' %zip_code
+                }
+                return render_template('index.html', **context)
+            
+            context = getContext(lat, lng, False, timestamp)
+
+            data = {'zip_code': zip_code, 'context': context}
+
+            def save_data(data):
+                if cache.get(CACHE_1_ZIP) is None:
+                    cache.set(CACHE_1_ZIP, data)
+                    return
+                elif cache.get(CACHE_2_ZIP) is None:
+                    cache.set(CACHE_2_ZIP, data)
+                    return
+                elif cache.get(CACHE_3_ZIP) is None:
+                    cache.set(CACHE_3_ZIP, data)
+                    return
```

Comment:
please separate cache logic into a separate PR/branch! That way we can focus on just the zip codes.